core/graph/src/test/java/uk/gov/gchq/gaffer/graph/GraphTest.java

```diff
@@ -1092,6 +1050,23 @@ public void shouldThrowExceptionIfStoreClassPropertyIsNotSet() throws OperationE
         }
     }
 
+    @Test
+    public void shouldThrowExceptionIfGraphIdIsMissing() {
+        final StoreProperties storeProperties = new StoreProperties();
+        storeProperties.setStoreClass(TestStoreImpl.class.getName());
+        try {
```

Comment:
Using assertJ would be better here for checking exception is called, as you should only be checking the `build()` method. Doing something like the following would be much better:

```java
Graph.Builder initNoGraphId = new Graph.Builder()
     .config(new GraphConfig.Builder()
         .build())
     .addSchema(new Schema())
     .storeProperties(new StoreProperties());

 assertThatExceptionOfType(IllegalArgumentException.class)
     .isThrownBy(initNoGraphId::build)
     .withMessageContaining("graphId is required");
```